src/AnimateText.js

```diff
@@ -0,0 +1,96 @@
+import React, { useCallback, useEffect, useRef, useState } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import SplitText from '@moxy/react-split-text';
+import IntersectionObserver from '@researchgate/react-intersection-observer';
+import 'intersection-observer'; // Polyfill to support older browsers
+
+const AnimateText = ({
+    children,
+    className,
+    separator,
+    withObserver,
+    shouldShow,
```

Comment:
Do we have any guideline on how these kind of props should be named, e.g. `show` vs `shouldShow`?

Reply:
Since I extracted this from a project I just went with the same. Probably `isShowing` would be better here? What do you think?

Reply:
I prefer just `show`, similarly to the attributes of DOM elements, e.g., `disabled`, `checked`,  `checked`, rather than `isDisabled`, etc.. However, more importantly than my personal preference is consistency with other libraries we have and which path we usually/should take. Maybe we should discuss this with the team.

Reply:
I am fine with `show` but you're right, I think that it is something that we can discuss with the team to keep the consistency amongst all the libraries.

Reply:
@acostalima I get your point of the DOM attributes and it's a good one, but in React I'm not a fan of using `show` or `disabled` as the type of value it holds is not explicit, it can either be a boolean or an object as far as I know. With `shouldShow`, `isDisabled`, etc. it's more explicit that it holds a boolean.

Reply:
@fsdiogo Well, in this particular case, where `show` is both a verb and a name, I'd never say `show` is an object. A function? More likely. My point here is that I've seen developers doing your way and other's mine. We should put our personal preferences aside and reach a consensus as a team. I don't want to block the PR because of this, but please start a discussion on Slack. When we have a decision, the repository's maintainer (@jdiogopeixoto) should follow up by opening an issue or PR to address the issue, if applicable. 💪 
Sounds good?

Reply:
I personally do not like `shouldShow` prop name, but it is functional and works. Personally I prefer something like `isVisible` but to be honest I don't really see the problem here. As long as the prop name states what it does or represents it in a clear way I'm fine with either name. I think we don't need to be that picky.

Reply:
I understand what @fsdiogo is saying that a good rule of thumb for naming booleans is to prefix them with `is*` or `should*`, but in this case we can argue that `show` is a good exception to this rule considering the context of this package which is to show or hide the text by animating it. If the package had more complex options the naming of the props might have to follow the rule to avoid ambiguous names. But in this case I think it's fine to just use `show` - you'd first assume it's a boolean considering you are using this package to show and hide text.

I don't think we need to apply the same naming rule to every single package, but rather find a common ground on what makes sense to be the same or different.

Let me know what you think.

Reply:
@zebateira sounds reasonable!